### DISCUSSION TITLE
Runner does not install docker-compose and aws-cli

### DISCUSSION THREAD

**Discussion by galoniz** [actions/actions-runner-controller#1885] [Questions] [answered]
Trying to run AWS commands and docker-compose commands on my runners. 
unfortunately I find a diff between the Github hosted-runners and the k8s self-hosted runners

I have installed the runners exactly like the guide in the README file

`Run aws s3 cp s3://webdriverio-files ./media --recursive
  aws s3 cp s3://webdriverio-files ./media --recursive
  shell: /usr/bin/bash -e {0}
  env:
    AWS_DEFAULT_REGION: us-east-1
    AWS_REGION: us-east-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
/runner/_work/_temp/df03a503-3c34-4b67-850d-73b94[2]
aca[4]00.sh: line 1: aws: command not found
Error: Process completed with exit code 12[7]`

and the same error goes for docker-compose

Is there a better image for those kind of operations?
Is there an option to add an init script to install them?

Your help is much appreciated here :)

**Comment by toast-gear** [comment, marked as answer]
https://github.com/actions-runner-controller/actions-runner-controller/blob/master/docs/detailed-docs.md#software-installed-in-the-runner-image you need to build your own image with any custom tools you need like the aws cli.

**Reply by trent-abc** [reply]
Docs were changed, assuming you are pointing to this section now - https://github.com/actions/actions-runner-controller/blob/master/docs/deploying-alternative-runners.md

**Comment by trent-abc** [comment]
Another option for future readers, is to manually install each time, I got something similar to the following working for me:
```
name: get_pods

on:
  workflow_dispatch:

jobs:
  release:
    runs-on: mylabel-runnerdeploy
    steps:
      - uses: actions/checkout@v2
      - name: Get pods
        shell: bash
        run: |
          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
          chmod +x ./kubectl
          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
          unzip awscliv2.zip
          sudo ./aws/install
          KUBECONFIG=some_config.yaml ./kubectl get pods --namespace "some-namespace" --context "some-context"
```
